solr/core/src/java/org/apache/solr/handler/admin/api/CoreReplicationAPI.java

```diff
@@ -68,6 +71,45 @@ public FileListResponse fetchFileList(
     return doFetchFileList(gen);
   }
 
+  @GET
```

Comment:
This is all correct and in line with the v2 APIs you did back in 2023 (PR #1620 PR #1704).

But we've made a slight tweak since then to support the [OpenAPI code generation stuff](https://issues.apache.org/jira/browse/SOLR-16825).  In short - we needed the JAX-RS annotations to live somewhere that gets compiled *before* "core", so that "core" can use all the nice generated code.  We managed this by creating a new Gradle module - "api" - where the JAX-RS annotations live on interfaces.  (Code in "core" can then implement those interfaces with methods containing the API logic).

Long story short - we'll want to do the same here.  There's some instructions [here](https://github.com/apache/solr/blob/main/dev-docs/apis.adoc#writing-jax-rs-apis) that might be helpful.  Though in your case most of the steps are already done - the pieces just need moved around a bit.

(Slightly my fault for not calling out more explicitly on SOLR-16470 that the setup had changed since your earlier commits, so I'm willing to make the necessary changes here.  You may beat me to it, but if not I'll try and update this eventually.)

Reply:
Ah good to know. No worries though, I just got back from vacation so will find some bandwidth soon to make the necessary adjustments.

Reply:
OK, I'll leave you to it, but if you have any questions or get stuck, lmk and I'll try to help out!

(Hope you had a great vacation!)